app/components/common/NicknameEditor.tsx

```diff
@@ -0,0 +1,113 @@
+'use client';
+
+/**
+ * Simple modal for editing wallet address nicknames.
+ * Nicknames are stored in browser localStorage.
+ */
+
+import React, { useState, useEffect } from 'react';
+import { getNickname, setNickname, removeNickname } from '@utils/nicknames';
+
+type Props = {
+    address: string;
+    onClose: () => void;
+};
+
+export function NicknameEditor({ address, onClose }: Props) {
```

Comment:
If you choose to move files to a separate feature directory, that is also a candidate to move there.

Also it would be great to have a Storybook's story for that component

Reply:
Created `app/features/nicknames/ui/stories/NicknameEditor.stories.tsx` with three stories:
- Primary (new nickname)
- WithExistingNickname (edit existing)
- LongAddress (edge case)